### ISSUE TITLE
"The file doesn't look right" error when using JSHint v2.7.0+

### ISSUE THREAD

**Issue by p0lar-bear** [luka-zitnik/jshint-for-netbeans#5]
I downloaded the latest (2.9.1) version of JSHint from [here](https://github.com/jshint/jshint/releases/tag/2.9.1) and pointed Netbeans to it, and it tells me that the file doesn't look right.

**Comment by luka-zitnik** [comment]
JSHint broke support for the Rhino JavaScript engine that this plugin uses. From JSHint version 2.7.0, file jshint.js can work when patched, like this one, http://download-cf.jetbrains.com/idea/jshint/jshint-2.7.0.js.

https://github.com/jshint/jshint/issues/2308

**Comment by p0lar-bear** [comment]
Specifically, it looks like the patch made to that was to add this block of code after the version comment:

```
if (typeof java !== 'undefined' &&
    typeof java.lang !== 'undefined' &&
    typeof java.lang.System !== 'undefined') {
    // rhino environment detected
    if (typeof global === 'undefined' && typeof self === 'undefined' && typeof window === 'undefined') {
        global = this;
    }
}
else if (typeof window === 'undefined') {
    window = {};
}
```

...and then to remove the `if (typeof window === 'undefined') window = {};` that follows.

Maybe we could edit the Readme to include this info? Alternatively, I haven't developed plugins for NetBeans before so I'm not sure how, but maybe we could also look into using a different JavaScript engine like V8?

**Comment by luka-zitnik** [comment]
All true. The patch is simple. It boils down to changing line 3 of jshint.js from `if (typeof window === 'undefined') window = {};` to `if (typeof window === 'undefined') window = this;`, if it is going to be run in Rhino environment only. That can be turned into an instruction of the readme. Alternatively, another JS engine can be used. Nashorn seems most natural. Yet another option is to set up the Rhino environment in the plugin so to substitute for the patch. All that if JSHint should really forget about Rhino.

Both, promoting a patch and engine switching, don't seem fun enough if there's low interest in making this plugin work with newer versions of JSHint. You know how to fix this for yourself, so let's wait some more.

**Comment by p0lar-bear** [comment]
Fair enough; my case for requiring a higher version of JSHint was because I was using options in my .jshintrc that 2.6.3 didn't recognize. I hadn't used JSHint before now so when I read the docs on the website, I was avoiding using deprecated options (namely, using the "esversion" option instead of "es3," "es5," or "esnext"). I'm fairly sure at the moment that I'm representing a fraction-percentile, people that need the patch will find this info here, and be savvy enough to do it themselves.

I think this should remain open, though, so I'm not closing it.

**Comment by luka-zitnik** [comment]
Sure, let's just keep this open for now.
